inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp

```diff
@@ -14,12 +13,38 @@ namespace vpu {
 
 namespace {
 
-class StaticShapeNMS final : public NonMaxSuppression {
+class StaticShapeNMS final : public StageNode {
 private:
     StagePtr cloneImpl() const override {
         return std::make_shared<StaticShapeNMS>(*this);
     }
 
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder>& orderInfo) override {
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement>& stridesInfo) override {
+    }
+
+    void finalizeDataLayoutImpl() override {
+    }
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport>& batchInfo) override {
+    }
+
+    StageSHAVEsRequirements getSHAVEsRequirementsImpl() const override {
+        // Current NMS implementation doesn't allow calculation of `> boxesThreshold` boxes using one SHAVE
```

Comment:
Should there be a check that `maxBoxesNum` could be fit into memory for maximum amount of shaves?

Reply:
Actually I have just moved this code and haven't modified it. Previously the `StaticShapeNMS` was a child of older `NonMaxSuppression`, but as `NonMaxSuppression` is not used now I have moved its methods to `StaticShapeNMS`.
BTW, this behavior will be fixed soon in a separate request as a part of `StaticShapeNMS` optimization.

Reply:
Ok, but please make sure in final version code is fixed